src/ManagerV2/SharedPersonalDataManager.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace App\ManagerV2;
+
+use App\Entity\Attributes\SharedDocument;
+use App\Entity\Attributes\SharedFolder;
+use App\Entity\Attributes\SharedPersonalData;
+use App\Entity\Document;
+use App\Entity\DonneePersonnelle;
+use App\Entity\Dossier;
+use App\Factory\SharedPersonalDataFactory;
+use App\Repository\UserRepository;
+use App\ServiceV2\Mailer\MailerService;
+use App\ServiceV2\Traits\SessionsAwareTrait;
+use App\ServiceV2\Traits\UserAwareTrait;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class SharedPersonalDataManager
+{
+    use UserAwareTrait;
+    use SessionsAwareTrait;
+
+    public function __construct(
+        private readonly EntityManagerInterface $em,
+        private readonly SharedPersonalDataFactory $factory,
+        private readonly MailerService $mailerService,
+        private readonly UserRepository $userRepository,
+        private readonly RequestStack $requestStack,
+        private readonly Security $security,
+    ) {
+    }
+
+    public function generateSharedPersonalDataAndSendEmail(DonneePersonnelle $personalData, string $email, string $locale = 'fr'): void
+    {
+        $user = $this->getUser();
+        if (!$user || !$personalData->getBeneficiaire()) {
+            $this->addFlashMessage('danger', 'user_not_found');
+
+            return;
+        }
+        $sharedPersonalData = $this->factory->generateSharedPersonalData($user, $personalData, $email, $locale);
+        $this->mailerService->sendSharedDocumentLink($sharedPersonalData, $email);
+
+        $this->addFlashMessage('success', 'share_document_success');
```

Comment:
No flash message generation in service method please :pray:

Reply:
If you want to check if user or beneficiary exists, then throw an exception and handle it when calling this method in the controller

Reply:
Yes I didn't spend much time on refactoring old implementation, but you right